src/editors/containers/TextEditor/hooks.js

```diff
@@ -1,115 +1,54 @@
-    externalUrl: imgHTML.src,
-    altText: imgHTML.alt,
-    width: imgHTML.width,
-    height: imgHTML.height,
-  });
-  openModal();
-};
-
 export const addImageUploadBehavior = ({ openModal, setImage }) => (editor) => {
-  editor.ui.registry.addButton('imageuploadbutton', {
+  editor.ui.registry.addButton(tinyMCE.buttons.imageUploadButton, {
     icon: 'image',
     onAction: openModal,
   });
-  editor.ui.registry.addButton('editimagesettings', {
+  editor.ui.registry.addButton(tinyMCE.buttons.editImageSettings, {
     icon: 'image',
-    onAction: module.openModalWithSelectedImage(editor, setImage, openModal),
+    onAction: module.openModalWithSelectedImage({ editor, setImage, openModal }),
   });
 };
 
-export const initializeEditorRef = (setRef, initializeEditor) => (editor) => {
-  setRef(editor);
-  initializeEditor();
-};
-
-// for toast onClose to avoid console warnings
-export const nullMethod = () => {};
-
-export const pluginConfig = {
-  plugins: StrictDict({
-    link: 'link',
-    codesample: 'codesample',
-    emoticons: 'emoticons',
-    table: 'table',
-    charmap: 'charmap',
-    code: 'code',
-    autoresize: 'autoresize',
-    image: 'image',
-    imagetools: 'imagetools',
-  }),
-  menubar: false,
-  toolbar: StrictDict({
-    do: 'undo redo',
-    formatselect: 'formatselect',
-    wieght: 'bold italic backcolor',
-    align: 'alignleft aligncenter alignright alignjustify',
-    indents: 'bullist numlist outdent indent ',
-    imageupload: 'imageuploadbutton',
-    link: 'link',
-    emoticons: 'emoticons',
-    table: 'table',
-    codesample: 'codesample',
-    charmap: 'charmap',
-    removeformat: 'removeformat',
-    hr: 'hr',
-    code: 'code',
-  }),
-  imageToolbar: StrictDict({
-    editImageSettings: 'editimagesettings',
-  }),
-};
-export const getConfig = (key) => {
-  if (key === 'imageToolbar' || key === 'toolbar') {
-    return Object.values(module.pluginConfig[key]).join(' | ');
-  }
-  return Object.values(module.pluginConfig[key]).join(' ');
-};
-
 export const editorConfig = ({
   setEditorRef,
   blockValue,
   openModal,
   initializeEditor,
   setSelection,
-  lmsEndpointUrl,
 }) => ({
-  onInit: (evt, editor) => module.initializeEditorRef(setEditorRef, initializeEditor)(editor),
+  onInit: (evt, editor) => {
+    setEditorRef(editor);
+    initializeEditor();
+  },
   initialValue: blockValue ? blockValue.data.data : '',
   init: {
     setup: module.addImageUploadBehavior({ openModal, setImage: setSelection }),
-    plugins: module.getConfig('plugins'),
-    menubar: false,
-    toolbar: module.getConfig('toolbar'),
-    imagetools_toolbar: module.getConfig('imageToolbar'),
-    imagetools_cors_hosts: [lmsEndpointUrl], // as image assets come from lms, we need to whitelist it.
-    height: '100%',
-    min_height: 500,
-    branding: false,
+    plugins: pluginConfig.plugins,
+    imagetools_toolbar: pluginConfig.imageToolbar,
+    toolbar: pluginConfig.toolbar,
+    contextmenu: 'link table',
```

Comment:
Is this the change that makes it so we don't get the right click menu on images?

Reply:
yep